api/pkg/utils/encryptionutils.go

```diff
@@ -0,0 +1,78 @@
+package utils
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+)
+import log "github.com/sirupsen/logrus"
+
+func EncryptWithAES256RandomKey(data []byte, key []byte) (error, []byte) {
+	// use the key, get the cipher
+	cipherBlock, cipherErr := aes.NewCipher(key)
+	if cipherErr != nil {
+		log.Errorf("Encryption error, cipher not generated")
+		return cipherErr, nil
+	}
+	// use the cipher block to encrypt
+	// gcm or Galois/Counter Mode, is a mode of operation
+	// for symmetric key cryptographic block ciphers
+	// https://en.wikipedia.org/wiki/Galois/Counter_Mode
+	aesgcm, gcmErr := cipher.NewGCM(cipherBlock)
+	if gcmErr != nil {
+		log.Errorf("Encryption error, GCM not generated")
+		return gcmErr, nil
+	}
+	nonce := make([]byte, aesgcm.NonceSize())
```

Comment:
Should we comment that the NonceSize is default 12 bytes

Reply:
added